alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -44,134 +50,70 @@ AgentId BasicBehaviour::getOwnId() const
-{
-    if (!isExecutingInContext()) {
-        return;
-    }
-    auto prev = _behResult.exchange(BehResult::FAILURE);
-    if (prev != BehResult::FAILURE) {
-        _engine->editPlanBase().addFastPathEvent(_execContext.load());
-        if (_trace) {
-            _trace->setTag("Result", "Fail");
-        }
-    }
-}
-
 bool BasicBehaviour::isTriggeredRunFinished()
 {
     return !_triggeredJobRunning.load();
 }
 
 void BasicBehaviour::doInit()
 {
-    assert(_behResult.load() == BehResult::UNKNOWN);
-    ++_execState;
-
-    if (!isExecutingInContext()) {
-        return;
-    }
-
-    // There is a possible race condition here in the sense that the _execState can be behind the _signalState
-    // and yet this behaviour can execute in the _signalState's RunningPlan context. However this is harmless
-    // since all methods are guarded by isExecutingInContext() which will return false in all such cases.
-    // Atomically set the signal context to nullptr so that the RunningPlan instance can be deleted
-    // when the behaviour is terminated
-    _execContext = _signalContext.exchange(nullptr);
-    initTrace();
     try {
-        traceInit("Behaviour");
         initialiseParameters();
     } catch (const std::exception& e) {
         ALICA_ERROR_MSG("[BasicBehaviour] Exception in Behaviour-INIT of: " << getName() << std::endl << e.what());
     }
-
-    _initExecuted.store(true);
-
-    // Do not schedule repeatable run job when behaviour is event driven or when frequency is 0.
-    if (!isEventDriven() && _msInterval > AlicaTime::milliseconds(0)) {
-        // TODO: account for delayed start
-        _activeRunJobId = _engine->editScheduler().schedule(std::bind(&BasicBehaviour::runJob, this, nullptr), getInterval());
-    }
 }
 
-void BasicBehaviour::runJob(void* msg)
+void BasicBehaviour::doRun()
 {
-    // TODO: get rid of msg
     try {
-        traceRun();
-        run(msg);
+        run(nullptr);
     } catch (const std::exception& e) {
         std::string err = std::string("Exception caught:  ") + getName() + std::string(" - ") + std::string(e.what());
         sendLogMessage(4, err);
     }
     _triggeredJobRunning = false;
 }
 
-void BasicBehaviour::doTrigger()
-{
-    if (!_behaviour->isEventDriven() || !isTriggeredRunFinished()) {
-        return;
-    }
-    _triggeredJobRunning = true;
-    _engine->editScheduler().schedule(std::bind(&BasicBehaviour::runJob, this, nullptr));
-}
-
 void BasicBehaviour::doTerminate()
 {
-
-    if (_activeRunJobId != -1) {
-        _engine->editScheduler().cancelJob(_activeRunJobId);
-        _activeRunJobId = -1;
-    }
     // Just to be double safe in terms of the correct behaviour of isSuccess() & isFailure() ensure result is reset before incrementing _execState
     _behResult.store(BehResult::UNKNOWN);
-    if (setTerminatedState()) {
-        return;
-    }
 
     // Intentionally call onTermination() at the end. This prevents setting success/failure from this method
```

Comment:
this is no longer applicable & I don't see any harm in setting success/failure in onTermination now

Reply:
there is no point in setting success or failure in ontermination as engine has already decided to shutdown the behaviour. Better utility is for isSuccess() and isFailure() methods. This information would be available in onTermination method now.